File: Lab2/apz-pzpi-22-1-tokar-denys-lab2-code/src/models/Rule.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const ruleSchema = new Schema({
    greenhouseId: { type: Schema.Types.ObjectId, ref: 'Greenhouse', required: true },
    condition: { type: String, required: true, enum: ['sensor_based'] },
    action: {
      type: String,
      required: true,
    },
    threshold: {
        sensorModelId: { type: String, required: true },
        operator: { type: String, required: true, enum: ['>', '<', '>=', '<='] },
        value: { type: Schema.Types.Mixed, required: true }
    },
    status: { type: String, enum: ['active', 'inactive'], default: 'active' },
});

module.exports = mongoose.model('Rule', ruleSchema);

File: Lab4/apz-pzpi-22-1-tokar-denys-lab4-code/app/src/main/java/com/example/apz_pzpi_22_1_tokar_denys_lab4_code/data/model/Rule.kt
package com.example.apz_pzpi_22_1_tokar_denys_lab4_code.data.model

data class Threshold(
    val sensorModelId: String,
    val operator: String,
    val value: Any
)

data class Rule(
    val _id: String,
    val greenhouseId: String,
    val condition: String, // 'sensor_based'
    val action: String,
    val threshold: Threshold,
    var status: String // 'active' or 'inactive' - var, щоб можна було змінювати локально
)